docs/source/format/CanonicalExtensions.rst

```diff
@@ -251,6 +251,24 @@ Variable shape tensor
    Values inside each **data** tensor element are stored in row-major/C-contiguous
    order according to the corresponding **shape**.
 
+.. _json_extension:
+
+JSON
+====
+
+* Extension name: `arrow.json`.
+
+* The storage type of this extension is ``StringArray`` or
+  ``StringViewArray``.
+
+* Extension type parameters:
+
+  This type does not have any parameters.
+
+* Description of the serialization:
+
+  Metadata is an empty string.
```

Comment:
Perhaps we could have a mechanism to encode JSON schema or similar? :thinking:

Reply:
That would be interesting, though I'm not sure about adding that until we have someone who will definitely implement using it.

We could phrase this in a way that metadata is currently optional. Maybe:

```suggestion
  Metadata is either an empty string or a JSON string with an empty object.
  In the future, additional fields may be added, but they are not required
  to interpret the array.
```

Reply:
Year, that makes sense as we are promising [backwards compatibility](https://arrow.apache.org/docs/format/CanonicalExtensions.html#making-modifications):

> Making Modifications
> Like standard Arrow data types, canonical extension types should be considered stable once standardized. Modifying a canonical extension type (for example to expand the set of parameters) should be an exceptional event, follow the same rules as laid out above, and provide backwards compatibility guarantees.